inst/include/dust/dust_cpu.hpp

```diff
@@ -437,10 +436,16 @@ class dust_cpu {
       }
       n = p.back().size(); // ensures all particles have same size
     }
-    // TODO: I think that we should disalow this bit of logic and
-    // number of particles should be surely fixed at initial creation?
-    // See #305
-    if (particles_.size() == n_particles_total_) {
+
+    if (particles_.empty()) {
```

Comment:
Does this new initialisation change things on the GPU at all?

Reply:
We never hold a vector on the gpu, and the current checks are better (they were what revealed this issue). However, they have more complex problems in #310